conda/cli/main_search.py

```diff
@@ -107,182 +126,97 @@ def configure_parser(sub_parsers):
 
 
 def execute(args, parser):
+    from .common import (ensure_override_channels_requires_channel,
+                         ensure_use_local)
+    from ..core.index import get_index
+    from ..models.match_spec import MatchSpec
+    from ..models.version import VersionOrder
     from ..base.context import context
-    from ..core.index import get_channel_priority_map
-    from ..exceptions import PackagesNotFoundError, ResolvePackageNotFound
 
-    try:
-        execute_search(args, parser)
-    except ResolvePackageNotFound as e:
+    platform = args.platform or ''
+    if platform and platform != context.subdir:
+        args.unknown = False
```

Comment:
Move line 142 to line 135.

    spec = MatchSpec(args.match_spec)

Then, starting at line 136, should look something liek

```
    spec = MatchSpec(args.match_spec)
    if spec.get_exact_value('subdir'):
        platform = spec.get_exact_value('subdir')
    elif args.platform:
        platform = args.platform
    else:
        platform = ''
    if platform and platform != context.subdir:
        args.unknown = False
    ensure_use_local(args)
    ensure_override_channels_requires_channel(args, dashc=False)
```